core/src/main/java/eu/drus/jpa/unit/decorator/jpa/PersistenceContextDecorator.java

```diff
@@ -31,7 +31,7 @@ public void beforeTest(final TestMethodInvocation invocation) throws Exception {
         if (field.getType().equals(EntityManager.class)) {
             final EntityManager em = getEntityManager(context, emf);
             context.storeData(Constants.KEY_ENTITY_MANAGER, em);
-            injectValue(field, invocation.getTestInstance(), em);
+            injectValue(field, invocation.getTestInstance().get(), em);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)

Reply:
if the required object is not present an exception has to be thrown anyway.